src/read.rs

```diff
@@ -255,3 +267,29 @@ where
         .unwrap_or_else(|| Collector::from_iter(iter::empty()))
     }
 }
+
+#[cfg(test)]
+mod test {
+    use crate::new;
+
+    #[test]
+    fn reserve_and_fit() {
+        let (r, mut w) = new();
+
+        // 64 is the second power of two higher than 17,
+        // so it should at least shrink the vector to less than
+        // or equal to the next highest power of 2 after 17
```

Comment:
Where does 17 come from?

Reply:
Ohh, because of how many things we insert further down, I see..

Reply:
I think it'd be useful to add something like
```rust
// 1 + 2^4, so it's a tiny bit more than a power of two
let n = 17;
```
at the top of the test, and then use `n` instead of a literal `17` elsewhere.